File: tests/maven/java/src/main/java/dev/morphia/critter/test/Generics.java
package dev.morphia.critter.test;

import dev.morphia.annotations.Entity;

import java.util.List;
import java.util.TreeMap;

@Entity
public class Generics {
    List<List<List<Address>>> listListList;
    TreeMap<String, List<Address>> mapList  = new TreeMap<>();

    public Generics(List<List<List<Address>>> listListList, TreeMap<String, List<Address>> mapList) {
        this.listListList = listListList;
        this.mapList = mapList;
    }

    public Generics(Address address) {
        mapList.put("1", List.of(address));
        listListList = List.of(List.of(List.of(address)));
    }

    public List<List<List<Address>>> getListListList() {
        return listListList;
    }

    public void setListListList(List<List<List<Address>>> listListList) {
        this.listListList = listListList;
    }

    public TreeMap<String, List<Address>> getMapList() {
        return mapList;
    }

    public void setMapList(TreeMap<String, List<Address>> mapList) {
        this.mapList = mapList;
    }
}

File: tests/maven/kotlin/src/main/kotlin/dev/morphia/critter/test/Generics.kt
package dev.morphia.critter.test

import dev.morphia.annotations.Entity
import java.util.TreeMap

@Entity
class Generics() {
    constructor(address: Address) : this() {
        mapList["1"] = listOf(address)
        listListList = mutableListOf(mutableListOf(listOf(address)))
    }

    var listListList: List<List<List<Address>>> = mutableListOf()
    var mapList: TreeMap<String, List<Address>> = TreeMap()
}